sdk/storage/AzureStorageBlob/Source/StorageBlobClient.swift

```diff
@@ -344,41 +339,111 @@ public class StorageBlobClient: PipelineClient, PagedCollectionDelegate {
+            try rawDownload(blob: blob, fromContainer: container, withOptions: options, then: completion)
         } else {
             // TODO: Test and reconsider this implemenation for the public URL scenario.
             let client = try StorageBlobClient(
                 accountUrl: host,
                 credential: credential,
                 withOptions: self.options
             )
-            try client.download(blob: blob, fromContainer: container, withOptions: options, then: completion)
+            try client.rawDownload(blob: blob, fromContainer: container, withOptions: options, then: completion)
+        }
+    }
+
+    public func download(
+        blob: String,
+        fromContainer container: String,
+        withOptions options: DownloadBlobOptions? = nil
+    ) throws -> Transfer? {
+        guard let context = manager.persistentContainer?.viewContext else { return nil }
+        let start = Int64(options?.range?.offset ?? 0)
+        let end = Int64(options?.range?.length ?? 0)
+        let downloader = try BlobStreamDownloader(
+            client: self,
+            delegate: nil,
+            name: blob,
+            container: container,
+            options: options
+        )
+        let blobTransfer = BlobTransfer.with(
+            context: context,
+            baseUrl: baseUrl,
+            blobName: blob,
+            containerName: container,
+            uri: downloader.downloadDestination,
+            startRange: start,
+            endRange: end,
+            parent: nil
+        )
+        blobTransfer.downloader = downloader
+        manager.add(transfer: blobTransfer)
+        return blobTransfer
+    }
+
+    public func download(url: URL, withOptions options: DownloadBlobOptions? = nil) throws -> Transfer? {
+        let (host, container, blob) = try parse(url: url)
+        if baseUrl == host {
+            return try download(blob: blob, fromContainer: container, withOptions: options)
+        } else {
+            let client = try StorageBlobClient(
+                accountUrl: host,
+                credential: credential,
+                withOptions: self.options
+            )
+            return try client.download(blob: blob, fromContainer: container, withOptions: options)
         }
     }
 
-    /**
-     Create a simple URL for a blob.
-     - Parameter blob: Name of the blob.
-     - Parameter container: Name of the container.
-     - Returns: The URL of the blob.
-     */
+    /// Create a simple URL for a blob.
+    /// - Parameters:
+    ///   - blob: The name of the blob.
+    ///   - container: The name of the container.
     public func url(forBlob blob: String, inContainer container: String) -> URL? {
         var url = URL(string: baseUrl)
         url?.appendPathComponent(container)
         url?.appendPathComponent(blob)
         return url
     }
+
+    public func localUrl(
+        blob: String,
```

Comment:
Should these have the same names as the parameters to `url()`? E.g. "forBlob", "inContainer"

Reply:
OR the others should match this scheme. :)

Reply:
Either way, we should use a preposition before the noun e.g. "forBlob"